docs/design/ovs-pipeline.md

```diff
@@ -333,58 +382,195 @@ more information on connection tracking in OVS.
+* Drop packets reported as invalid by conntrack.
+* Forward non-Service tracked packets to table [EgressRuleTable] directly.
+* Forward untracked packet to tables [SessionAffinityTable], [ServiceLBTable].
 
 If you dump the flows for this table, you should see the following:
 
 ```text
-1. table=31, priority=190,ct_state=+inv+trk,ip actions=drop
-2. table=31, priority=0 actions=goto_table:40
+1. table=31, priority=200,ct_state=-new+trk,ct_mark=0x21,ip actions=load:0x1->NXM_NX_REG0[19],resubmit(,50)
+2. table=31, priority=190,ct_state=+inv+trk,ip actions=drop
+3. table=31, priority=190,ct_state=-new+trk,ip actions=resubmit(,50)
+4. table=31, priority=0 actions=resubmit(,40),resubmit(,41)
 ```
 
-Flow 1 drops invalid traffic. All non-dropped traffic finally goes to the
-[DNATTable].
+Flow 1 is used to forward tracked Service packets to table [EgressRuleTable] directly,
+without passing table [SessionAffinityTable], [ServiceLBTable], [EndpointDNATTable].
+The flow also sets bit 19 in NXM_NX_REG0 to 0x1, which indicates that the destination
+and source MACs of the packet should be rewritten in the [l3ForwardingTable].
 
-### DNATTable (40)
+Flow 2 is used to drop packets which is reported as invalid by conntrack.
 
-At the moment this table's only job is to send traffic destined to Services
-through the local gateway, without any modifications. kube-proxy will then take
-care of load-balancing the connections across the different backends for each
-Service.
+Flow 3 is used to forward tracked non-Service packets to [EgressRuleTable] directly,
+without passing table [SessionAffinityTable], [ServiceLBTable], [EndpointDNATTable].
 
-If you dump the flows for this table, you should see something like this:
+Flow 4 is used to match the first packet of untracked connection and forward it to table
+[SessionAffinityTable], [ServiceLBTable].
+
+### SessionAffinityTable (40)
+
+If `service.spec.sessionAffinity` of a Service is "None", this table will set the value
+of bits [16..18] in NXM_NX_REG4 to 0b001, which indicates that the Service needs to do
+Endpoint selection. If you dump the flow, you should see the flow like following:
+
+```text
+table=40, priority=0 actions=load:0x1->NXM_NX_REG4[16..18]
+```
+
+If `service.spec.sessionAffinity` of a Service is "ClientIP", when a client accesses
+the Service for the first time, a learned flow with hard timeout which equals
+`service.spec.sessionAffinityConfig.clientIP.timeoutSeconds` of the Service will be
+generated in this table. This will be explained in detail in chapter table [ServiceLBTable].
+
+### ServiceLBTable (41)
+
+This table is used to implement Service Endpoint selection. Note that, currently, only
+ClusterIP whose client is from the cluster is supported, which means that client should
+be Pods. NodePort, LoadBalancer and ClusterIP whose client is from K8s Node will be
+supported in the future.
+
+When a ClusterIP Service whose `service.spec.sessionAffinity` is None is created, if you
```

Comment:
When a ClusterIP Service is created with `service.spec.sessionAffinity` set to `None`,

Reply:
Thanks, much better.